### PR TITLE
Sdit 763 prison incentive changed

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesReferenceService.kt [hunk 1]
```diff
@@ -48,6 +61,21 @@ class IncentivesReferenceService(
...
+  private fun trackEvent(eventName: String, incentiveLevel: PrisonIncentiveLevel) {
+    telemetryClient.trackEvent(
+      eventName,
+      mutableMapOf(
+        "code" to incentiveLevel.levelCode,
+        "prison" to incentiveLevel.prisonId,
+        "defaultOnAdmission" to incentiveLevel.defaultOnAdmission.toString(),
+        "active" to incentiveLevel.active.toString(),
+        "visitOrderAllowance" to incentiveLevel.active.toString(),
+        "visitOrderAllowance" to incentiveLevel.active.toString(),
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesReferenceService.kt:73]: copy paste error?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesReferenceService.kt [hunk 2]
```diff
@@ -60,11 +88,38 @@ class IncentivesReferenceService(
...
+      incentivesApiService.getPrisonIncentiveLevel(prisonId, incentiveLevelCode)
+        .also { prisonIncentiveLevelData ->
+          nomisApiService.updatePrisonIncentiveLevel(prisonId, prisonIncentiveLevelData.toNomisPrisonIncentiveLevel())
+          trackEvent("prison-incentive-level-updated", prisonIncentiveLevelData)
+        }
+    } ?: also {
+      incentivesApiService.getPrisonIncentiveLevel(prisonId, incentiveLevelCode)
+        .also { prisonIncentiveLevelData ->
+          nomisApiService.createPrisonIncentiveLevel(prisonId, prisonIncentiveLevelData.toNomisPrisonIncentiveLevel())
+          trackEvent("prison-incentive-level-inserted", prisonIncentiveLevelData)
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesReferenceService.kt:104]: Do we need some telemetry if the API calls fails for some reason?

**Review by mikehalmamoj — DISMISSED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesReferenceService.kt (hunk 1 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesReferenceService.kt:73]: sure is - thanks

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesReferenceService.kt (hunk 2 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesReferenceService.kt:108]: I don't think it adds much in this simpler process (no mapping calls). I might come back to it after testing on dev.

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**